### PR TITLE
Update search location drawing tools for 2D map

### PR DESCRIPTION
**Pull request by jrnorth**
* Search location shapes can now be edited on the 2D map.
* A new drawing toolbar is now displayed at the top of the map.
* The 'cancel drawing' banner has been removed. The 'Draw' button now toggles to 'Cancel Drawing' when clicked.
* Fixes some bugs with how lines and polygons that cross the antimeridian are displayed.

### Testing Instructions
* Test both the basic and advanced search menus.
* Test drawing and editing all 4 shapes (polygon, bounding box, line, point-radius).
  * Drawing: draw a completely new shape.
  * Editing: edit the points/boundaries of an existing shape, and add existing points.
    * Line: add new points by clicking and dragging existing line segments, and move any points on the line.
    * Point-radius: move the center point, and change the radius by moving the circle's edge.
    * Bounding box: move any of the corners or edges.
    * Polygon: add new points by clicking and dragging existing line segments, and move any points on the line.
* Test entering shape coordinates manually, then editing the resulting shape.
* Test drawing and editing shapes that cross the antimeridian.
  * There are some limitations here, ask me for details.
* Test cancelling the drawing via the cancel button on the toolbar and the draw button on the search menu that is toggled to a cancel button when a drawing is in progress.
* Perform some basic testing on the 3D map to ensure drawing was not broken.

https://user-images.githubusercontent.com/4495447/224818911-15d8639a-900e-4a5d-992f-afba748d4276.mp4

### CODE REVIEW

**Comment by jrnorth** [comment]
build now

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/openlayers/drawing-and-display.tsx [hunk 1]
```diff
@@ -13,18 +13,219 @@
...
+import Common from '../../../../js/Common'
+import wreqr from '../../../../js/wreqr'
+import { Drawing } from '../../../../component/singletons/drawing'
+import * as Turf from '@turf/turf'
+import { Position, Point, Polygon, LineString } from '@turf/turf'
+import { validateGeo } from '../../../../react-component/utils/validation'
+import ShapeUtils from '../../../../js/ShapeUtils'
+import _ from 'lodash'
+
+// TODO should move all the common code to ddf-ui and have downstream import it
```
Comment (alexaabrd) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/openlayers/drawing-and-display.tsx:47]: ✏️  Remove this TODO

Comment (alexabird) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/openlayers/drawing-and-display.tsx]: ❓ should this TODO be removed?

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/openlayers/drawing-and-display.tsx [hunk 2]
```diff
@@ -38,6 +239,259 @@ export const removeOldDrawing = ({ map, id }: { map: ol.Map; id: string }) => {
...
+    case 'Point Radius':
+      return createPointRadiusModel(geo)
+    case 'Bounding Box':
+      return createBoundingBoxModel(geo)
+    default:
+      return {}
+  }
+}
+
+// TODO do we need to do any polygon 'fixing' here? Or does the validation take care of that?
```
Comment (alexabird) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/openlayers/drawing-and-display.tsx]: Assuming validation covers issues needing fixing

**Review by alexabird — APPROVED** [review]

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/openlayers/drawing-and-display.tsx (hunk 1 above)
Comment (jrnorth) [reply ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/openlayers/drawing-and-display.tsx]: This is still valid and I'm addressing it now.

**Comment by jrnorth** [comment]
Will post some videos demonstrating the changes later today.

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/openlayers/drawing-and-display.tsx (hunk 2 above)
Comment (jrnorth) [reply ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/openlayers/drawing-and-display.tsx]: Yes, removed this.

**Comment by jrnorth** [comment]
@alexaabrd All TODOs have been resolved.

**Review by alexabird — APPROVED** [review]

**Review by pvargas — APPROVED** [review]

**Comment by alexabird** [comment]
Went through all the testing steps and looks good! 
✏️ Came across:
1. When in "Draw" mode, I wasn't able to click on "Layout" to open the 2D map (I think this was an issue before, but not as noticeable since "Layout" was covered by "Cancel Drawing")
2. "Cancel Drawing" becomes an option for all anyGeo filters even though only 1 is being canceled.

**Comment by jrnorth** [comment]
@alexabird Fixed issue 2, if you want to retest just that.

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/react-component/location/location.tsx [hunk 3]
```diff
@@ -207,20 +208,36 @@ const LocationInput = ({ onChange, value }: any) => {
...
-              color="primary"
-              fullWidth
-            >
-              <span className="fa fa-globe" />
-              <span className="ml-2">Draw</span>
-            </Button>
+            isDrawing ? (
+              <Button
+                className="location-draw mt-2"
+                onMouseDown={() => {
```
Comment (andrewkfiedler) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/react-component/location/location.tsx]: No clue why the old code did onMouseDown, but can we update these to onClick?

**Review by andrewkfiedler — APPROVED** [review]
Noticed that the buffering from the display to the drawing doesn't quite match up, but not sure why / which is more correct.

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/react-component/location/location.tsx (hunk 3 above)
Comment (jrnorth) [reply ui-frontend/packages/catalog-ui-search/src/main/webapp/react-component/location/location.tsx]: Done.

**Comment by jrnorth** [comment]
build now

**Pull request merged**